runtimes/eoapi/stac/eoapi/stac/app.py

```diff
@@ -142,3 +144,75 @@ async def viewer_page(request: Request):
         },
         media_type="text/html",
     )
+
+
+if settings.jwks_url:
```

Comment:
can we move this to a `auth.py` with a method like `def add_auth(app: FastAPI, settings: ApiSettings):` 

and in `app.py` we will just do 
```python
if settings.jwks_url:
    add_auth(api, settings)
```